src/data/nat/digits.lean

```diff
@@ -456,6 +461,44 @@ begin
   exact base_pow_length_digits_le' b m,
 end
 
+lemma of_digits_zeros {α : Type*} [semiring α] (b : α) {L : list ℕ} (heq : ∀ a ∈ L, a = 0) :
+  of_digits b L = 0 :=
+begin
+  induction L with d L ih,
+  { exact of_digits_nil b, },
+  { rw [of_digits, heq d (list.mem_cons_self d L), cast_zero, zero_add],
+    refine mul_eq_zero_of_right b (ih _),
+    exact λ a' ah, heq a' (list.mem_cons_of_mem d ah), }
+end
+
+/--
+Applying `(digits b ∘ of_digits b)` will create a prefix of the original list if it has proper
+digits; it removes trailing zeros.
+-/
+lemma digits_of_digits_prefix  {b : ℕ} {L : list ℕ} (hb : 2 ≤ b) (hlt : ∀ a ∈ L, a < b) :
```

Comment:
Are you sure you need `hb`?

Reply:
Oh, I thought I needed it. I've removed it now.